addons/relate/ftests/src/main/java/org/commonjava/indy/relate/ftest/RelDownloadPomNotFound404Test.java

```diff
@@ -0,0 +1,65 @@
+package org.commonjava.indy.relate.ftest;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.commonjava.indy.ftest.core.AbstractIndyFunctionalTest;
+import org.commonjava.indy.model.core.RemoteRepository;
+import org.commonjava.test.http.expect.ExpectationServer;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.InputStream;
+
+import static org.commonjava.indy.model.core.StoreType.remote;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Created by ruhan on 2/17/17.
+ */
+public class RelDownloadPomNotFound404Test
+                extends AbstractIndyFunctionalTest
+{
+    private static final String path = "org/foo/bar/1/bar-1.pom";
+
+    private static final String pathRel = path + ".rel";
+
+    private static final String content =
+                    "<project><modelVersion>4.0.0</modelVersion><groupId>org.foo</groupId><artifactId>bar</artifactId><version>1</version></project>";
+
+    @Rule
+    public ExpectationServer server = new ExpectationServer();
+
+    @Test
+    public void run() throws Exception
+    {
+        final String repo1 = "repo1";
+
+        server.expect( server.formatUrl( repo1, path ), 404, "not found" );
+
+        RemoteRepository remote1 = new RemoteRepository( repo1, server.formatUrl( repo1 ) );
+        client.stores().create( remote1, "adding remote", RemoteRepository.class );
+
+        // Download .rel before even touching POM
+        InputStream rel = client.content().get( remote, repo1, pathRel );
+        assertThat( rel, nullValue() );
+
+        logger.debug( ">>> register again ..." );
+        server.expect( server.formatUrl( repo1, path ), 200, content ); // upstream restored
```

Comment:
Where does this nfc come in during this test scenario? I searched and found many places using it. It will be very helpful if there is any design doc about nfc.